src/dune_file.ml

```diff
@@ -641,35 +641,28 @@ module Auto_format = struct
   let syntax =
     Syntax.create ~name:"fmt"
       ~desc:"integration with automatic formatters"
-      [ (1, 1) ]
+      [ (1, 2) ]
 
   type language =
-    | Ocaml
-    | Reason
+    | Dialect of string
     | Dune
 
   let language_to_dyn =
     let open Dyn.Encoder in
     function
-    | Ocaml -> constr "ocaml" []
-    | Reason -> constr "reason" []
+    | Dialect name -> constr "dialect" [string name]
     | Dune -> constr "dune" []
 
-  let language =
-    sum
-      [ ("ocaml", return Ocaml)
-      ; ("reason", return Reason)
-      ; ("dune",
-         let+ () = Syntax.since syntax (1, 1) in
-         Dune)
-      ]
+  let language = function
+    | "dune" -> Dune
+    | s -> Dialect s
 
   type enabled_for =
     | Default of Syntax.Version.t
     | Only of language list
 
   let enabled_for_field =
-    let+ r = field_o "enabled_for" (repeat language)
+    let+ r = field_o "enabled_for" (repeat (map ~f:language string))
```

Comment:
We do not validate that the language names used in the `(enabled_for ...)` field actually exist (i.e. are either `dune` or a dialect name). This validation has to happen after the `dune-project` file has been parsed, but the "extensions" API would need to be extended to allow doing such a thing.